desc/magnetic_fields/_core.py

```diff
@@ -238,23 +256,91 @@ def compute_Bnormal(
             the locations (in specified basis) at which the Bnormal was calculated
 
         """
+        calc_Bplasma = False
         if isinstance(surface, EquilibriaFamily):
             surface = surface[-1]
         if isinstance(surface, Equilibrium):
-            surface = surface.surface
+            calc_Bplasma = True
+            eq = surface
+            surface = eq.surface
         if eval_grid is None:
             eval_grid = LinearGrid(
                 rho=jnp.array(1.0), M=2 * surface.M, N=2 * surface.N, NFP=surface.NFP
             )
+        if vc_source_grid is None:
+            source_NFP = surface.NFP if surface.N > 0 else 64
```

Comment:
Question on this, when N of the eq is zero, we make the virtual casing source grid NFP be 64. But if the eq N is 0 then so is the eq N_grid right? so how is it, for example in the tutorial axisymmetric free boundary solve, that the toroidal effects are accounted for if the source grid has no toroidal grid nodes?

Reply:
there's always still nodes at zeta=0, and NFP=64 means we repeat this 64 times around the torus. There's no toroidal coupling though so each of those can be done independently.